week4-task/src/api/user.ts

```diff
@@ -0,0 +1,49 @@
+import { apiClient } from "./apiClient";
+import axios from "axios";
+import { ApiResponse } from "../types/user.types";
+import { getNetworkError } from "@utils/apiError";
+
+export const getMyNickname = async (): Promise<ApiResponse> => {
+  const userId = localStorage.getItem("userId");
+
+  try {
+    const response = await apiClient.get<ApiResponse>("/api/v1/users/me", {
+      headers: { userId },
+    });
+    return response.data;
+  } catch (error) {
+    if (axios.isAxiosError(error) && error.response && error.response.data) {
+      return error.response.data as ApiResponse;
+    }
+    return getNetworkError<ApiResponse>();
+  }
+};

```

Comment:
```suggestion
function handleError<T>(error: unknown): T {
  if (apiClient.isAxiosError(error) && error.response && error.response.data) {
    return error.response.data as T;
  }
  return getNetworkError<T>();
}

export const getMyNickname = async (): Promise<ApiResponse> => {
  const userId = localStorage.getItem("userId");

  try {
    const response = await apiClient.get<ApiResponse>("/api/v1/users/me", {
      headers: { userId },
    });
    return response.data;
  } catch (error: unknown) {
    return handleError<ApiResponse>(error);
  }
};
```
전체적으로 에러 핸들링 로직이 반복되고 있는데, 분리해주는건 어떨까요?